### ISSUE TITLE
dvc cache status: slow multiple files

### ISSUE THREAD

**Issue by dmpetrov** [dataversioncontrol/dvc#474]
```
$ dvc cache status
(1/7): [########################################] 100% dog.168.jpg
(2/7): [########################################] 100% dog.173.jpg
(3/7): [########################################] 100% dog.174.jpg
(4/7): [########################################] 100% dog.181.jpg
(5/7): [########################################] 100% dog.19.jpg
(6/7): [########################################] 100% dog.194.jpg
(7/7): [########################################] 100% dog.198.jpg
(8/7): [########################################] 100% dog.205.jpg
(9/7): [########################################] 100% dog.238.jpg
[SKIP 500 files overall]
	new file:	.dvc/cache/fd47d523fa82bb9c282b897b55be0442
	new file:	.dvc/cache/0dd437f2f04fbf54355beab021494baa
	new file:	.dvc/cache/730786e5c78eb2782124e998c802ae9c
	new file:	.dvc/cache/0f69da47a6a31007964f838477825473
```

It is very slow to check all the files. Any chance to improve the performance?

**Comment by efiop** [comment]
Need to reuse s3 and gcp connections when processing multiple files.

**Comment by efiop** [comment]
Fixed by #504.
